src/manageUsers.py

```diff
@@ -0,0 +1,232 @@
+#ipoptester13@ejabberd : None
+#ipoptester14@ejabberd : None
+#ipoptester15@ejabberd : None
+#ipoptester16@ejabberd : None
+#ipoptester17@ejabberd : None
+'''
+
+
+import sys
+import sleekxmpp
+import argparse
+import ConfigParser
+import shelve
+import random
+import logging
+
+'''Handle character encoding if using sub 3 python'''
+if sys.version_info <(3,0):
+    reload(sys)
+    sys.setdefaultencoding('utf8')
+
+''' This class interacts with XMPP server, logs into the server
+with room admin credentials and invites/blocks other users
+Mapping between JID and IP address allocated to them are maintained in 
+a file database.'''
+class roomManager(sleekxmpp.ClientXMPP):
+
+    def __init__(self,jid,password,room_name,action,jid_file):
+        super(roomManager,self).__init__(jid,password)
+        self.room_name = room_name
+        self.nick = jid.split("@")[0]
+        users = shelve.open('ip_table.db')
+        self.jid_file = jid_file
+        self.ip_table = dict(users)
+        users.close()
+        self.action = action
+        self.add_event_handler('session_start', self.session_start)
+        self.add_event_handler('muc::%s::got_online' % self.room_name, \
+                                self.muc_online)
+        
+    def session_start(self, event):
+        self.send_presence()
+        self.get_roster()
+        self.room = self.plugin['xep_0045']
+        self.room.joinMUC(self.room_name,self.nick)
+        
+        
+    def muc_online(self, presence):
+
+        logging.debug("presence message : %s"%(presence))
+        if presence['muc']['nick'] == self.nick:
+            logging.debug("ip table: %s"%(self.ip_table))
+            '''action here refers to the presence of invite
+            flag in command line argument,depending on which either
+            a invitation is sent to the JID's in applicants.ini file
+            or their access to the room is revoked.'''
+            if self.action != None:
+                for ip in self.ip_table:
+                    self.room.invite(self.room_name,self.ip_table[ip],\
+                        reason = '#'+ip+'#')
+            elif self.action == None:
+                for user in self.jid_file:
+                    self.room.setAffiliation(self.room_name, \
+                        jid=user,affiliation='none')
+        
+            self.send_presence('offline')
+            self.disconnect(wait=True)
+'''Randomly generate a IP address, checks if it is unassigned
+and allocate it to a new user.also make a note of it in the database.
+currently generates a address in the range (ip prefix && 255.255.xxx.xxx).
+can be modified to accomdate any other range.'''            
+def allocateIP(prefix,user_jid):
+    parts = prefix.split(".")
+    ip_prefix = parts[0] + "." + parts[1] + "."
+    ip_table = shelve.open('ip_table.db',writeback=True)
+    jid_ip_table = shelve.open('jid_ip_table.db',writeback=True)
+    ip_prefix = ip_prefix + str(random.randint(0,255)) + "." + \
```

Comment:
IP address ending 0 is not used. 
Also IP address ending 255 is reserved for broadcast address. 
So randint range should be 1, 254